gpii/node_modules/eventLog/src/eventLog.js

```diff
@@ -263,11 +297,90 @@ gpii.eventLog.writeLog = function (that, level, event) {
     event.level = intLevel.value;
 
     // Log to console before the installation ID and timestamp are added (no one wants to see it).
-    fluid.log(fluid.logLevel.IMPORTANT, event);
+    fluid.log(fluid.logLevel.TRACE, event);
 
     event.installID = that.installationID;
     event.timestamp = gpii.eventLog.getTimestamp();
-    fs.appendFileSync(that.logFilePath, JSON.stringify(event) + "\n");
+    event.version = that.version;
+
+    var logLine = JSON.stringify(event) + "\n";
+
+    if (that.logHost && that.logPort) {
+        gpii.eventLog.writeLogTcp(that, logLine);
+    }
+
+    if (that.logFilePath) {
+        fs.appendFileSync(that.logFilePath, logLine);
+    }
+};
+
+/**
+ * Sends a log line to the configured log server.
+ *
+ * @param {Component} that - The gpii.eventLog instance.
+ * @param {String} logLine - The log line, including the newline.
+ */
+gpii.eventLog.writeLogTcp = function (that, logLine) {
+    if (!that.logSocket) {
+        gpii.eventLog.connectLog(that);
+    }
+
+    if (that.logSocket.connecting) {
+        // Buffer the output until the connection is made. Socket.write already buffers, however if the connection
+        // fails the data will be lost.
+        if (!that.logBuffer) {
```

Comment:
Can't see any convincing reason for this to be lazy, make it a member instead?

Reply:
me neither